cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java

```diff
@@ -91,9 +100,24 @@ public static void init() throws Exception {
     cConf.set(Constants.CFG_LOCAL_DATA_DIR, TEMP_FOLDER.newFolder().getAbsolutePath());
     logBaseDir = cConf.get(LoggingConfiguration.LOG_BASE_DIR);
     namespacesDir = cConf.get(Constants.Namespace.NAMESPACES_DIR);
+    final LocalLocationFactory localLocationFactory =
+      new LocalLocationFactory(new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR)));
     injector = Guice.createInjector(
       new ConfigModule(cConf, hConf),
-      new NonCustomLocationUnitTestModule().getModule(),
+      // use HDFS modules to simulate error cases (local module does not throw the same exceptions as HDFS)
+      Modules.override(new LocationRuntimeModule().getDistributedModules()).with(
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            // Override namespace location factory so that it does not perform lookup for NamespaceMeta like
+            // DefaultNamespacedLocationFactory does and hence allow unit
+            // tests to use it without creating namespace meta for the namespace.
+            // This is similar to what NonCustomLocationUnitTestModule does.
+            bind(NamespacedLocationFactory.class).to(NamespacedLocationFactoryTestClient.class);
+            bind(LocationFactory.class).toInstance(localLocationFactory);
```

Comment:
Might be useful to put a comment that we need this to make LocationFactory use the dir for this junit test.